### PR TITLE
Added environment variables & secrets

### PR DESCRIPTION
**Pull request by Froxz**
Added environment:

[Secrets](https://docs.github.com/en/rest/actions/secrets?apiVersion=2022-11-28#list-environment-secrets)
[Variables](https://docs.github.com/en/rest/actions/variables?apiVersion=2022-11-28#list-environment-variables)

### CODE REVIEW

**Comment by Froxz** [comment]
Hi @acrobat when have time please review 👍

File: lib/Github/Api/Environment/Secrets.php [hunk 1]
```diff
@@ -0,0 +1,91 @@
...
+     */
+    public function createOrUpdate(int $id, string $name, string $secretName, array $parameters = [])
+    {
+        if (!isset($parameters['encrypted_value'])) {
+            throw new MissingArgumentException(['encrypted_value']);
+        }
+
+        if (!isset($parameters['key_id'])) {
+            throw new MissingArgumentException(['key_id']);
+        }
```
Comment (acrobat) [inline lib/Github/Api/Environment/Secrets.php:60]: I would not add this validation in our code, but let the github api validate the request and return any errors related to missing parameters. This way we won't block user in the future if github changes the required parameters

File: lib/Github/Api/Environment/Variables.php [hunk 2]
```diff
@@ -0,0 +1,94 @@
...
+     */
+    public function create(int $id, string $name, array $parameters)
+    {
+        if (!isset($parameters['name'])) {
+            throw new MissingArgumentException(['name']);
+        }
+
+        if (!isset($parameters['value'])) {
+            throw new MissingArgumentException(['value']);
+        }
```
Comment (acrobat) [inline lib/Github/Api/Environment/Variables.php:59]: I would not add this validation in our code, but let the github api validate the request and return any errors related to missing parameters. This way we won't block user in the future if github changes the required parameters

**Review by acrobat — CHANGES_REQUESTED** [review]

**Review by acrobat — APPROVED** [review]

**Comment by acrobat** [comment]
Thank you @Froxz

**Comment by Froxz** [comment]
@acrobat Thank you.

**Pull request merged**